### PR TITLE
fix(query): type runtime-validated angular-query responses with the schema's zod output alias

### LINKED ISSUE #3941: angular-query: runtimeValidation types the validated response as zod.input, not zod.output
Sibling of #3938, which reports this for the fetch client. The angular-query client has the same mismatch.

## What happens

With `output.schemas.type: 'zod'` and `override.query.runtimeValidation` enabled, the generated request function pipes the response through `Schema.safeParse` and returns `result.data`, so the runtime value is the schema's output type. The declared TypeScript type is still the schema name, and in zod-schemas mode that name is aliased to `zod.input<typeof Schema>` (`packages/orval/src/write-zod-specs.ts:244`).

The snapshot shows it. `tests/__snapshots__/runtime-validation/angular-query-both/endpoints.ts` declares `listPets(...): Promise<Pets>` and `http.get<Pets>(...)` while the rxjs `map` returns the parsed output. With `coerce` or `useDates`, input and output diverge, so the caller sees the pre-coercion type where the runtime value is a `Date`.

## Why the Angular clients don't have this

#3650 centralized the value side of runtime validation into `emitResponseValidation` (`packages/core/src/generators/runtime-validation.ts`). The type side, referencing the generated `XOutput = zod.output<typeof X>` aliases, was only implemented in `packages/angular`: `getSchemaOutputTypeRef` in `packages/angular/src/utils.ts:65`, applied in `http-client.ts` and `http-resource.ts`. The angular-query path in `packages/query/src/client.ts:249` uses the shared emitter but never switches the declared type.

The fetch client (#3938) is missing the same half.

## Proposed fix

Move the type-side helpers into core next to the emitter, so the declared type and the emitted validation can't drift:

- `getSchemaOutputTypeRef`, `isPrimitiveType`, `hasSchemaImport` from `packages/angular/src/utils.ts`
- a helper that resolves the declared response type: the output ref when validation applies to a json response with a schema import, the original name otherwise
- a shared import rewrite (flip the schema import to `values: true`, add the `XOutput` type import). This is duplicated in `packages/fetch/src/index.ts:825`, `packages/query/src/index.ts:104` and `packages/angular/src/http-client.ts:900`, and the first two omit the `XOutput` import.

One wrinkle: `packages/query/src/client.ts:247` renames `Error` to `ErrorSchema` for the value ref, so the helper needs the schema value name and the type name as separate inputs.

Suggested order: core helper plus the fetch fix (#3938) first, then this issue, then refactor `packages/angular` onto the core helpers with byte-identical snapshots.

### PR DESCRIPTION
**Pull request by the-ult**
Sibling of #3943, which fixed this for the fetch client. With `runtimeValidation` on a zod-schemas output, the angular-query request function pipes the response through `Schema.safeParse` and returns `result.data`, so the runtime value is the schema's `zod.output` type. The declared TypeScript type still named the schema, which zod-schemas mode aliases to `zod.input`. Under `coerce`, `useDates`, defaults or transforms the two diverge: the runtime value is a `Date`, the declared type says `string | Date | unknown`.

The `HttpClient` generic and the `Promise` return type now reference the generated `XOutput` aliases, matching the angular `HttpClient` client:

```ts
// before
export const listPets = (...): Promise<Pets> => { ... http.get<Pets>(url) ... }
// after
export const listPets = (...): Promise<PetsOutput> => { ... http.get<PetsOutput>(url) ... }
```

The generated hooks pick the type up through `Awaited<ReturnType<typeof listPets>>`, so no hook code changes. Operations with a custom mutator keep the schema (input) type — the generated parse never runs there — and non-angular query flavors are byte-identical.

Both changes ride on the core helpers #3943 introduced:

- `packages/query/src/client.ts` resolves the declared type via `getSchemaOutputTypeRef` and the value ref via `getSchemaValueRef` (the `Error` → `ErrorSchema` rename, covered by a unit test asserting `ErrorSchema.parse` next to `Promise<ErrorOutput>`).
- `packages/query/src/index.ts` replaces its hand-rolled schema-import flip with `rewriteImportsForResponseValidation`, which also adds the missing `XOutput` type import. Adapter creation moves above the rewrite so `adapter.isAngularHttp` can gate the alias import to angular-query.

Review order: `packages/query/src/client.ts`, `packages/query/src/index.ts`. The snapshot changes (`runtime-validation/angular-query-both` plus the `samples/angular-query` zod sample) are mechanical: `Promise<X>` → `Promise<XOutput>` and the added type imports. Docs gain the same output-alias note the fetch pages got in #3943.

Per #3941's suggested order, the remaining step is refactoring `packages/angular` onto the core helpers with byte-identical snapshots — not part of this PR.

Fixes #3941
Refs #3943

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01RdXNjYsbphgDxir14cZqXU

## Summary by CodeRabbit

* **New Features**
  * Improved Angular Query runtime validation with accurate Zod output types for validated responses.
  * Added support for imported response schemas and shared validation behavior.
  * Preserved input types for custom mutators and non-Zod or disabled validation scenarios.

* **Documentation**
  * Clarified runtime-validation typing behavior and updated Angular Query support guidance.

* **Tests**
  * Added coverage for validated responses, error schemas, and validation edge cases.

### CODE REVIEW

File: packages/query/src/index.ts [hunk 1]
```diff
@@ -89,40 +92,42 @@ export const generateQuery: ClientBuilder = async (
...
-
-  const adapter = createFrameworkAdapter({
-    outputClient,
-    packageJson: options.context.output.packageJson,
-    queryVersion: normalizedVerbOptions.override.query.version,
-  });
+    verbOptions.override.query.runtimeValidation?.enabled &&
+    isZodOutput &&
+    !isPrimitiveResponseType(responseType) &&
+    hasSchemaImport(verbOptions.response.imports, responseType);
```
[the remark this answers is not in the document]
Comment (the-ult) [reply packages/query/src/index.ts:112]: Accepted — fixed in cf44ae7cd. `shouldUseRuntimeValidation` now gates on `!verbOptions.mutator`, so mutator operations skip the import rewrite entirely and their schema import stays type-only (query has no `includeZodSchemaInArguments`, unlike fetch, so the value import served nothing there). Full fixture regeneration confirms zero snapshot changes — no existing config combines `query.runtimeValidation` + zod + mutator.

**Review by melloware — APPROVED** [review]
